src/Microsoft.DotNet.Darc/DarcLib/LocalLibGit2Client.cs

```diff
@@ -0,0 +1,553 @@
+    {
+        const string crlf = "\r\n";
+        const string lf = "\n";
+
+        // Check gitAttributes to determine whether the file has eof handling set.
+        var result = await _processManager.ExecuteGit(repoPath, new[] { "check-attr", "eol", "--", filePath });
+        result.ThrowIfFailed($"Failed to determine eol for {filePath}");
+
+        string eofAttr = result.StandardOutput.Trim();
+
+        if (string.IsNullOrEmpty(eofAttr) ||
+            eofAttr.Contains("eol: unspecified") ||
+            eofAttr.Contains("eol: auto"))
+        {
+            if (Environment.NewLine != crlf)
+            {
+                return content.Replace(crlf, Environment.NewLine);
+            }
+            else if (Environment.NewLine == crlf && !content.Contains(crlf))
+            {
+                return content.Replace(lf, Environment.NewLine);
+            }
+        }
+        else if (eofAttr.Contains("eol: crlf"))
+        {
+            // Test to avoid adding extra \r.
+            if (!content.Contains(crlf))
+            {
+                return content.Replace(lf, crlf);
+            }
+        }
+        else if (eofAttr.Contains("eol: lf"))
+        {
+            return content.Replace(crlf, lf);
+        }
+        else
+        {
+            throw new DarcException($"Unknown eof setting '{eofAttr}' for file '{filePath};");
+        }
+
+        return content;
+    }
+
+    /// <summary>
+    ///     Checkout the repo to the specified state.
+    /// </summary>
+    /// <param name="commit">Tag, branch, or commit to checkout.</param>
+    public void Checkout(string repoPath, string commit, bool force = false)
+    {
+        _logger.LogDebug($"Checking out {commit}", commit ?? "default commit");
+        var checkoutOptions = new CheckoutOptions
+        {
+            CheckoutModifiers = force ? CheckoutModifiers.Force : CheckoutModifiers.None,
+        };
+        try
+        {
+            _logger.LogDebug($"Reading local repo from {repoPath}");
+            using (var localRepo = new Repository(repoPath))
+            {
+                if (commit == null)
+                {
+                    commit = localRepo.Head.Reference.TargetIdentifier;
+                    _logger.LogInformation($"Repo {localRepo.Info.WorkingDirectory} default commit to checkout is {commit}");
+                }
+                try
+                {
+                    _logger.LogDebug($"Attempting to check out {commit} in {repoPath}");
+                    SafeCheckout(localRepo, commit, checkoutOptions);
+                    if (force)
+                    {
+                        CleanRepoAndSubmodules(localRepo);
+                    }
+                }
+                catch (NotFoundException)
+                {
+                    _logger.LogWarning($"Couldn't find commit {commit} in {repoPath} locally.  Attempting fetch.");
```

Comment:
This feels a bit wrong, like we're using try/catch for something that can be done with an if/else.
Maybe we should try to see if the repo contains the commit we're looking for with something like `git branch --contains <commit-id>`

Reply:
I only copied this over from the Helper class. I would not change functionality where I don't have to ideally

Reply:
Ahh I see, I guess it's better left alone for now at least